signers/legacy/search.go

```diff
@@ -216,6 +141,41 @@ func generateSignature(content string, request_time int64, secret string, nonce
 	return hmac_string
 }
 
+func (v *SearchSigner) ParseAuthHeaders(req *http.Request) map[string]string {
+	auth_headers := map[string]string{}
+	auth_fields := []string{
+		"acquia_solr_time",
+		"acquia_solr_nonce",
+		"acquia_solr_hmac",
+	}
+
+	for _, field_name := range auth_fields {
+		auth_cookie, err := req.Cookie(field_name)
+		if err != nil {
+			logger.Print("Error retrieving:", field_name)
+		} else {
+			auth_headers[field_name] = auth_cookie.Value
+		}
+	}
+	return auth_headers
+}
+
+func (v *SearchSigner) GetResponseSigner() signers.ResponseSigner {
+	return v.respSigner
+}
+
 func (v *SearchSigner) Version() int {
 	return 0
 }
+
+func (v *SearchSigner) HashBody(r *http.Request) (string, *signers.AuthenticationError) {
+	return "", nil
```

Comment:
For the functions we expect not to be used, should we be explicit and panic or do something?